Framework/Core/include/Framework/ASoA.h

```diff
@@ -1192,7 +1198,7 @@ struct TableIterator : IP, C... {
       [this]<typename T>(T*) -> void {},
     };
     (f(static_cast<C*>(nullptr)), ...);
-    if constexpr (has_index_v) {
```

Comment:
Actually we should probably move the part above to some "preBind" function with constraints on the type T.